vendor/github.com/rancher/rancher/pkg/auth/providers/common/usermanager.go

```diff
@@ -619,7 +619,18 @@ func userByPrincipal(obj interface{}) ([]string, error) {
 		return []string{}, nil
 	}
 
-	return u.PrincipalIDs, nil
+	match := false
+	for _, id := range u.PrincipalIDs {
+		if strings.HasPrefix(id, "local://") {
+			match = true
```

Comment:
Just trying to understand this part, would the logic change if we just return at this point?

Reply:
This got pulled in because of `go mod vendor`, my changes were for the `rancher/types` related stuff